app/scripts/models/flow.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent a login flow, holding the data that we
+ * need to submit as part of our metrics when we perform actions
+ * during the flow.
+ *
+ * Try to fetch the flowId and flowBegin from the resume token.
+ * If unavailable there, fetch from data attributes in the DOM.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var Backbone = require('backbone');
+  var Cocktail = require('cocktail');
+  var ErrorUtils = require('lib/error-utils');
+  var ResumeTokenMixin = require('models/mixins/resume-token');
+  var SearchParamMixin = require('models/mixins/search-param');
+  var vat = require('lib/vat');
+
+  var Model = Backbone.Model.extend({
+    initialize: function (options) {
+      options = options || {};
+
+      this.sentryMetrics = options.sentryMetrics;
+      this.window = options.window || window;
+
+      this.fetch();
+    },
+
+    defaults: {
+      flowBegin: null,
+      flowId: null
+    },
+
+    fetch: function () {
+
+      var error;
+      var $body = this.window['$']('body');
+
+      this.populateFromStringifiedResumeToken(this.getSearchParam('resume'));
+
+      // We should either get both fields from the resume token, or neither.
+      // Getting one without the other is an error.
+
+      if (this.get('flowId')) {
+        if (! this.get('flowBegin')) {
+          error = AuthErrors.toMissingResumeTokenPropertyError('flowBegin');
+          ErrorUtils.captureError(error, this.sentryMetrics);
+        }
+      } else if (this.get('flowBegin')) {
+        error = AuthErrors.toMissingResumeTokenPropertyError('flowId');
+        ErrorUtils.captureError(error, this.sentryMetrics);
+      } else {
```

Comment:
I dont think there's value in taking e.g. the `flowId` from one and the `flowBeginTime` from the other, because as you say, it'll just pollute the data with something that we know can't be right.  (My first version actually did this, I changed when I wrote the tests and realized that what I was testing didn't seem to be useful behaviour)